BD-4429: Updates to Connected Content docs (#9325)
braze-inc/braze-docs: _docs/_user_guide/personalization_and_dynamic_content/connected_content/caching_responses.md

---
nav_title: Caching Responses
article_title: Caching Connected Content Responses
page_order: 2.5
description: "This article covers how to cache Connected Content responses across different campaigns or messages in the same workspace to optimize sending speeds."
---

# Caching Connected Content responses

> Connected Content responses can be cached across different campaigns or messages (in the same workspace) to optimize send speeds.

> Braze doesn’t permanently log or store Connected Content responses. If you explicitly choose to store a Connected Content call response as a Liquid variable, Braze only stores this in-memory, meaning on temporary storage that’s deleted after a short period of time, to render the Liquid variable and send the message.

Reviewer: ```suggestion
Braze doesn’t permanently log or store Connected Content responses. If you explicitly choose to store a Connected Content call response as a Liquid variable, Braze only stores this in-memory, on temporary storage that’s deleted after a short period of time, to render the Liquid variable and send the message.
```

> Braze doesn’t permanently log or store Connected Content responses. If you explicitly choose to store a Connected Content call response as a Liquid variable, Braze only stores this in-memory, meaning on temporary storage that’s deleted after a short period of time, to render the Liquid variable and send the message.

Reviewer: ```suggestion
Braze doesn’t permanently log or store Connected Content responses. If you explicitly choose to store a Connected Content call response as a Liquid variable, Braze only stores this on temporary storage that’s deleted after a short period of time, to render the Liquid variable and send the message.
```

To prevent caching, you can specify `:no_cache`, which may cause increased network traffic. To help troubleshoot and monitor system health, Braze may also log Connected Content calls that fail (such as `404` and `429`). These logs are kept for up to 30 days.

## Default cache settings

> The cache age is up to 5 minutes (300 seconds). You can update this by adding the :cache_max_age parameter to the Connected Content call. An example is:

Reviewer: ```suggestion
The cache age is up to five minutes (300 seconds). You can update this by adding the `:cache_max_age` parameter to the Connected Content call. An example is:
```

{% raw %}
```
{{ {% connected_content [https://example.com/webservice.json] :cache_max_age 900 %}}}
```
{% endraw %}

GET requests are cached. You can configure this by adding the :no_cache parameter to the Connected Content call.

POST requests are not cached. This can be forced by adding the :cache_max_age parameter to the Connected Content call. The minimum cache time is 5 minutes, and the maximum cache time is 4 hours.

> Note that cache settings aren’t guaranteed. Caching can reduce calls to your endpoints, so we recommend using multiple calls per endpoint within the cache duration rather than being overly reliant on caching.

Reviewer: ```suggestion
{% alert note %}
Cache settings aren’t guaranteed. Caching can reduce calls to your endpoints, so we recommend using multiple calls per endpoint within the cache duration rather than being overly reliant on caching.
{% endalert %}
```
Suggestion for turning this into an alert!

### Cache size limit

> The Connected Content response body can be up to 1&nbsp;MB. If the response body is larger than 1&nbsp;MB, it will not cache.

Reviewer: ```suggestion
The Connected Content response body can be up to one&nbsp;MB. If the response body is larger than one&nbsp;MB, it will not cache.
```

## Cache time 

> Connected Content will cache the value it returns from GET endpoints for a minimum of 5 minutes. If a cache time is not specified, the default cache time is 5 minutes.

Reviewer: ```suggestion
Connected Content will cache the value it returns from GET endpoints for a minimum of five minutes. If a cache time is not specified, the default cache time is five minutes.
```

> Connected Content cache time can be configured to be longer with :cache_max_age, as shown in the following example. The minimum cache time is 5 minutes and the maximum cache time is 4 hours. Connected Content data is cached in-memory using a volatile cache system, such as Memcached. 

Reviewer: ```suggestion
Connected Content cache time can be configured to be longer with :cache_max_age, as shown in the following example. The minimum cache time is five minutes and the maximum cache time is four hours. Connected Content data is cached in-memory using a volatile cache system, such as Memcached. 
```
